pycrunch/shoji.py

```diff
@@ -247,3 +267,29 @@ def graph(self, newgraph):
         """Update the Order with the new graph."""
         self['graph'] = newgraph
         super(Order, self).put(data=self.json)
+
+
+class TaskProgressTimeoutError(Exception):
+    def __init__(self, entity, response):
+        super(TaskProgressTimeoutError, self).__init__(
+            'Task Progress did not complete before timeout. '
+            'Trap this exception and call exc.entity.wait_progress(exc.response) '
```

Comment:
See https://github.com/Crunch-io/pycrunch/blob/440a654681b817f3aa9073d319bdbadac711383e/pycrunch/progress.py it now provides a way to configure progress behaviour and reporting.

It can be provided both to Session to configure a global one, or to `.create`  to provide a specific one.

See https://github.com/Crunch-io/pycrunch/pull/19/files#diff-85056a6fb58e66a49d4e43b1723a7768R132 for an example on how it can be used.